rust_src/src/frames.rs

```diff
@@ -244,4 +244,31 @@ pub fn frame_pointer_visible_p(frame: LispObject) -> bool {
     !frame_ref.pointer_invisible()
 }
 
+/// Return the root window of FRAME-OR-WINDOW.
+/// If omitted, FRAME-OR-WINDOW defaults to the currently selected frame.
+/// With a frame argument, return that frame's root window.
+/// With a window argument, return the root window of that window's frame.
+#[lisp_fn(min = "0")]
+pub fn frame_root_window(frame_or_window: LispObject) -> LispObject {
+    let frame = window_frame_live_or_selected(frame_or_window);
+    frame.root_window
+}
+
+/* Don't move this to window.el - this must be a safe routine.  */
```

Comment:
Out of curiosity, what are reasons to port C functions to lisp ?

Reply:
I can see a desire by Emacs authors to have as little C as possible. As the Rust port shows, there is little to this function.